File: .tmp_sc4013/java_22/4000_largest_integer_with_given_digit_sum.java
// CONFIG class=Solution method=largestInteger types=None
// LeetCode 4000 - Largest Integer With Given Digit Sum
// https://leetcode.com/problems/largest-integer-with-given-digit-sum/

class Solution {
    public int largestInteger(int n, int s) {
        if (n * 9 < s) return -1;
        int ans = 0;
        for (int i = 0; i < n; i++) {
            int x = s < 9 ? s : 9;
            ans = ans * 10 + x;
            s -= x;
        }
        return ans;
    }
}

File: .tmp_rs4013/rust_check22/4000_largest_integer_with_given_digit_sum.rs
struct Solution;
// LeetCode 4000 - Largest Integer With Given Digit Sum
// https://leetcode.com/problems/largest-integer-with-given-digit-sum/

impl Solution {
    pub fn largest_integer(n: i32, mut s: i32) -> i32 {
        if n * 9 < s {
            return -1;
        }
        let mut ans = 0;
        for _ in 0..n {
            let x = if s < 9 { s } else { 9 };
            ans = ans * 10 + x;
            s -= x;
        }
        ans
    }
}

fn main() {}
